Document libzstd in INSTALL.adoc (#12574)
ocaml/ocaml: INSTALL.adoc

* If you do not have write access to `/tmp`, you should set the environment
  variable `TMPDIR` to the name of some other temporary directory.

> * The zstd library is used for compression of marshaled data, if it is not available
>   the option `--without-zstd` should be passed to `configure` disabling it.

Reviewer: This wording is a bit unclear in my opinion. If `zstd` is not available, then compression is automatically disabled. The `--without-zstd` switch is to be used to disable this support even if `zstd` _is_ available. I suggest
```suggestion
* The zstd library is used for compression of marshaled data. The option
  `--without-zstd` may be passed to `configure` in order to disable it.
```

Author: Yes that is more clear.
